restli-client/src/main/java/com/linkedin/restli/client/RestClient.java

```diff
@@ -636,6 +714,11 @@ private ContentType resolveContentType(MessageHeadersBuilder<?> builder, DataMap
         {
           header = contentType.getHeaderKey();
         }
+        else if (_experimentalContentType != null
+            && experimentalContentTypeHeaders.contains(_experimentalContentType.getHeaderKey()))
+        {
+          header = _experimentalContentType.getHeaderKey();
+        }
```

Comment:
This logic can be:

for header  in serverAnnouncedContentTypes {
  if (ContentType.get(header).isPresent()) {
    return ContentType.getRequestContentType(header, ..)
  }
}